bundle/endpoint/client_example_test.go

```diff
@@ -0,0 +1,89 @@
+package endpoint_test
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spiffe/go-spiffe/bundle"
+	"github.com/spiffe/go-spiffe/bundle/endpoint"
+)
+
+// This is an example of the bundle endpoint client used with the the SPIFFE
+// authentication mechanism.
+// The example assumes an SPIRE implementation of the SPIFFE bundle endpoint server
+// is running on localhost:8443 under the trust domain 'spiffe://example.org'.
+func Example_client() {
+	// Load an initial up-to-date bundle from the remote trust domain (obtained
+	// through an offline exchange).
+	rootCAs, err := loadCertificate("dummy_upstream_ca.crt")
+	if err != nil {
+		fmt.Printf("Unable to load root CAs: %v", err)
+		return
+	}
+
+	// Creates the bundle client using the SPIFFE authentication mechanim
+	c, err := endpoint.NewClient(endpoint.ClientConfig{
+		EndpointAddress: "localhost:8443",
+		SPIFFEAuth: &endpoint.SPIFFEAuthConfig{
+			EndpointSpiffeID: "spiffe://example.org/spire/server",
+			RootCAs:          []*x509.Certificate{rootCAs},
+		},
+	})
+
```

Comment:
keep the error check close to the NewClient invocation (i.e. remove the newline)